src/ContactFormlet/ContactFormlet.js

```diff
@@ -14,39 +15,77 @@ function ContactFormlet({contacts, contactType, handleContactChange, contactData
     }
   }
   const [isNew, setIsNew] = useState(false);
+  const [contact, setContact] = useState({name: '', phoneNumber: ''})
+  const [options, setOptions] = useState([])
 
-  const options = contacts.map(contact => {
+  const [mutationName] = useMutation(mutationQuery, {
+    refetchQueries: [
+      queryName
+    ]
+  });
+
+  const createOption = (contact) => {
     return (
       <option key={contact.id} value={contact.id}>{contact.name}</option>
     )
-  })
+  }
+
+  const { data } = useQuery(queryName, {
+    onCompleted: data => {
+      setOptions(() => {
+        return data[resType].map(contact => createOption(contact))
+      })
+    }})
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    mutationName({
+      variables: {input: {params: contact}}
```

Comment:
I definitely need to go over this and look at some more examples for the mutations.